From 6fc5d182f175eaabcf7cfbf545bbb645ff50d2ea Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Fri, 3 Jan 2020 14:33:40 +0100
Subject: [PATCH 6/6] TFLite: pip package: support cross compilation
 environment variables

Add build environment variable to allow to cross compile TensorFlow Lite
pip package for other platform than Rpi or X86.

Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
---
 .../lite/tools/pip_package/build_pip_package.sh   | 13 +++++++++++--
 tensorflow/lite/tools/pip_package/setup.py        | 15 +++++++++++++++
 2 files changed, 26 insertions(+), 2 deletions(-)

diff --git a/tensorflow/lite/tools/pip_package/build_pip_package.sh b/tensorflow/lite/tools/pip_package/build_pip_package.sh
index 488945cdf5..c01125cfcc 100755
--- a/tensorflow/lite/tools/pip_package/build_pip_package.sh
+++ b/tensorflow/lite/tools/pip_package/build_pip_package.sh
@@ -23,7 +23,11 @@ export TENSORFLOW_DIR="${SCRIPT_DIR}/../../../.."
 TENSORFLOW_LITE_DIR="${TENSORFLOW_DIR}/tensorflow/lite"
 TENSORFLOW_VERSION=$(grep "_VERSION = " "${TENSORFLOW_DIR}/tensorflow/tools/pip_package/setup.py" | cut -d= -f2 | sed "s/[ '-]//g")
 export PACKAGE_VERSION="${TENSORFLOW_VERSION}${VERSION_SUFFIX}"
-BUILD_DIR="/tmp/tflite_pip/${PYTHON}"
+if [ "${TENSORFLOW_CROSS_COMPILATION}" = "true" ]; then
+  BUILD_DIR="`dirname $0`/gen/tflite_pip"
+else
+  BUILD_DIR="/tmp/tflite_pip/${PYTHON}"
+fi
 
 # Build source tree.
 rm -rf "${BUILD_DIR}" && mkdir -p "${BUILD_DIR}/tflite_runtime"
@@ -49,7 +53,12 @@ case "${TENSORFLOW_TARGET}" in
                        bdist_wheel --plat-name=linux-aarch64
     ;;
   *)
-    ${PYTHON} setup.py bdist bdist_wheel
+    if [ -n "${TENSORFLOW_TARGET}" ] && [ -n "${TENSORFLOW_TARGET_ARCH}" ]; then
+      ${PYTHON} setup.py bdist --plat-name=${TENSORFLOW_TARGET}-${TENSORFLOW_TARGET_ARCH} \
+                         bdist_wheel --plat-name=${TENSORFLOW_TARGET}-${TENSORFLOW_TARGET_ARCH}
+    else
+      ${PYTHON} setup.py bdist bdist_wheel
+    fi
     ;;
 esac
 
diff --git a/tensorflow/lite/tools/pip_package/setup.py b/tensorflow/lite/tools/pip_package/setup.py
index 90416b77bc..598612daad 100644
--- a/tensorflow/lite/tools/pip_package/setup.py
+++ b/tensorflow/lite/tools/pip_package/setup.py
@@ -50,6 +50,21 @@ elif TARGET == 'aarch64':
   os.environ['CC'] = 'aarch64-linux-gnu-gcc'
 MAKE_CROSS_OPTIONS = ['TARGET=%s' % TARGET]  if TARGET else []
 
+TARGET_ARCH = (
+    os.environ['TENSORFLOW_TARGET_ARCH'] if 'TENSORFLOW_TARGET_ARCH' in os.environ
+    else None)
+MAKE_CROSS_OPTIONS += ['TARGET_ARCH=%s' % TARGET_ARCH]  if TARGET_ARCH else []
+
+CC_PREFIX = (
+    os.environ['TENSORFLOW_CC_PREFIX'] if 'TENSORFLOW_CC_PREFIX' in os.environ
+    else None)
+MAKE_CROSS_OPTIONS += ['CC_PREFIX=%s' % CC_PREFIX]  if CC_PREFIX else []
+
+EXTRA_CXXFLAGS = (
+    os.environ['TENSORFLOW_EXTRA_CXXFLAGS'] if 'TENSORFLOW_EXTRA_CXXFLAGS' in os.environ
+    else None)
+MAKE_CROSS_OPTIONS += ['EXTRA_CXXFLAGS=%s' % EXTRA_CXXFLAGS]  if EXTRA_CXXFLAGS else []
+
 RELATIVE_MAKE_DIR = os.path.join('tensorflow', 'lite', 'tools', 'make')
 MAKE_DIR = os.path.join(TENSORFLOW_DIR, RELATIVE_MAKE_DIR)
 DOWNLOADS_DIR = os.path.join(MAKE_DIR, 'downloads')
-- 
2.17.1

